### PR TITLE
Use libintl on musl when necessary

### PR DESCRIPTION
**Pull request by ericonr**
Based on #24030 

@sgn

### CODE REVIEW

File: srcpkgs/widelands/patches/musl.patch [hunk 1]
```diff
@@ -10,15 +12,17 @@
...
+-
 -  # OpenBSD needs the X11 include directory in order to find GL/glu.h
 -  if (CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
 -    find_package(X11 REQUIRED)
 -    include_directories(${X11_INCLUDE_DIR})
 -  endif ()
 -endif (CMAKE_SYSTEM_NAME MATCHES "FreeBSD" OR CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
++# Find libexecinfo and libintl for musl
++find_library(EXECINFO_LIBRARY NAMES execinfo)
++find_library(INTL_LIBRARY NAMES intl)
```
Comment (sgn) [inline srcpkgs/widelands/patches/musl.patch:24]: `find_package(Intl, REQUIRED)` would be better, I think.
I didn't dig into it, though.

Comment (sgn) [reply srcpkgs/widelands/patches/musl.patch:24]: Anyway, does:
```
LDFLAGS="-lexecinfo -lintl"
```
works? If yes, it would be easier for us than patching

Comment (ericonr) [reply srcpkgs/widelands/patches/musl.patch:24]: I tried first with only `-lintl` and CMake completely ignored it. That's why I changed the patch :/

**Comment by ericonr** [comment]
I will try to split the PR to let the CI finish. widelands is kind of stupid with this.

File: srcpkgs/widelands/patches/musl.patch (hunk 1 above)
Comment (sgn) [reply srcpkgs/widelands/patches/musl.patch:24]: CMake doesn't ignore it.
`cmake` just makes it hard to use correctly.
They pass `$LDFLAGS` before `*.a`, by order of filename, it couldn't file symbols in those libs.

`cmake` is `cmake`.

**Comment by sgn** [comment]
On 2020-08-05 16:29:41+0000, Érico Nogueira Rolim <notifications@github.com> wrote:
> I will try to split the PR to let the CI finish. widelands is kind of stupid with this.

Not necessary, I split it into four branches here:

https://travis-ci.org/github/sgn/void-packages/builds/715687552
https://travis-ci.org/github/sgn/void-packages/builds/715687616
https://travis-ci.org/github/sgn/void-packages/builds/715687706
https://travis-ci.org/github/sgn/void-packages/builds/715687792

-- 
Danh

**Pull request closed without merging**